zrml/court/src/tests.rs

```diff
@@ -93,6 +110,70 @@ fn on_dispute_stores_jurors_that_should_vote() {
+        setup_blocks(1..2);
+        Court::join_court(Origin::signed(ALICE)).unwrap();
+        Court::join_court(Origin::signed(BOB)).unwrap();
+        Court::join_court(Origin::signed(CHARLIE)).unwrap();
+        Court::on_dispute(BASE, &[], &0, &ALICE).unwrap();
+        Court::vote(Origin::signed(ALICE), 0, OutcomeReport::Scalar(1)).unwrap();
+        Court::vote(Origin::signed(BOB), 0, OutcomeReport::Scalar(1)).unwrap();
+        Court::vote(Origin::signed(CHARLIE), 0, OutcomeReport::Scalar(2)).unwrap();
+        let outcome = Court::on_resolution(&|_| 0, &[], &0, &DEFAULT_MARKET).unwrap();
+        assert_eq!(outcome, OutcomeReport::Scalar(1))
+    });
+}
+
+#[test]
+fn on_resolution_sets_late_jurors_as_tardy() {
+    ExtBuilder::default().build().execute_with(|| {
+        setup_blocks(1..2);
+        Court::join_court(Origin::signed(ALICE)).unwrap();
+        Court::join_court(Origin::signed(BOB)).unwrap();
+        Court::vote(Origin::signed(ALICE), 0, OutcomeReport::Scalar(1)).unwrap();
+        Court::on_dispute(BASE, &[], &0, &ALICE).unwrap();
+        let _ = Court::on_resolution(&|_| 0, &[], &0, &DEFAULT_MARKET).unwrap();
+        assert_eq!(Jurors::<Runtime>::get(ALICE).unwrap().status, JurorStatus::Ok);
+        assert_eq!(Jurors::<Runtime>::get(BOB).unwrap().status, JurorStatus::Tardy);
+    });
+}
+
+#[test]
+fn on_resolution_punishes_tardy_jurors() {
+    ExtBuilder::default().build().execute_with(|| {
+        setup_blocks(1..2);
+        Court::join_court(Origin::signed(ALICE)).unwrap();
+        Court::join_court(Origin::signed(BOB)).unwrap();
+        Court::set_juror_as_tardy(&BOB).unwrap();
+        Court::vote(Origin::signed(ALICE), 0, OutcomeReport::Scalar(1)).unwrap();
+        Court::on_dispute(BASE, &[], &0, &ALICE).unwrap();
+        let _ = Court::on_resolution(&|_| 0, &[], &0, &DEFAULT_MARKET).unwrap();
+        let slash = 8000000000;
+        assert_eq!(Balances::free_balance(TREASURY), 1_000 * BASE + slash);
+        let join_court_reserved = 40000000000;
+        assert_eq!(Balances::reserved_balance(BOB), join_court_reserved - slash);
+    });
+}
+
+#[test]
+fn on_resolution_removes_requested_jurors_and_votes() {
+    ExtBuilder::default().build().execute_with(|| {
+        setup_blocks(1..2);
+        Court::join_court(Origin::signed(ALICE)).unwrap();
+        Court::join_court(Origin::signed(BOB)).unwrap();
+        Court::join_court(Origin::signed(CHARLIE)).unwrap();
+        Court::on_dispute(BASE, &[], &0, &ALICE).unwrap();
+        Court::vote(Origin::signed(ALICE), 0, OutcomeReport::Scalar(1)).unwrap();
+        Court::vote(Origin::signed(BOB), 0, OutcomeReport::Scalar(1)).unwrap();
+        Court::vote(Origin::signed(CHARLIE), 0, OutcomeReport::Scalar(2)).unwrap();
+        let _ = Court::on_resolution(&|_| 0, &[], &0, &DEFAULT_MARKET).unwrap();
+        assert_eq!(RequestedJurors::<Runtime>::iter().count(), 0);
+        assert_eq!(Votes::<Runtime>::iter().count(), 0);
+    });
+}
```

Comment:
Would be nice to have a test for `set_jurors_that_sided_on_the_second_most_voted_outcome_as_tardy`.

Reply:
Done